File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
  public:
  vector<int> topKFrequent(vector<int>& nums, int k) {
    vector<pair<int,int>>p;
    unordered_map<int,int>mp;
    vector<int>ans;


    for(auto it : nums){
        mp[it]++;
    }

    for(auto it : mp){
        p.push_back({it.first , it.second});
    }

        sort(p.begin(), p.end(), [](const pair<int, int>& a, const pair<int, int>& b) {
            return a.second > b.second; // Sort by frequency (descending)
        });


    for(int i = 0; i < k ; i++){
       ans.push_back(p[i].first);
    }

   return  ans;
    
    
  }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # most obvious sol that came in my mind :) if interviewer has promised not to judge me!!
        mp = {}

        for num in nums:
            mp[num] = mp.get(num , 0) + 1
        
        sorted_mp = dict(sorted(mp.items() , key=lambda x: x[1] ,reverse=True))

        ans = []
        for K , v in sorted_mp.items():
            if len(ans) < k:
                ans.append(K)
        
        return ans

        
